conn/doc.go

```diff
@@ -4,6 +4,45 @@
 
 // Package conn defines core interfaces for protocols and connections.
 //
-// Subpackages implements all the interfaces that can be used to connects
-// peripherals.
+// Subpackages implements all the supported interfaces that can be used to
+// connects peripherals.
+//
+// Concepts
```

Comment:
I would suggest to add a line talking about the connection between the physical world and this package. A developer might have a board and sensor or LED and wants to write code to control the LED. She might have no understanding of what buses, port or communications formats are available but found this package that offers connections to hardware. How can we offer basic concepts so she can clearly map her knowledge with this package and understand where to maybe dig (learning new things or exploring the code)? While this might be considered outside of the scope of code documentation, it might be extremely useful to new users and shouldn't prevent experienced users to get to what they are looking for.